FantasmoSDK/Classes/Extensions/CLLocation+Extension.swift

```diff
@@ -0,0 +1,133 @@
+//
+//  CLLocation+Extension.swift
+//  FantasmoSDK
+//
+//  Created by lucas kuzma on 8/4/21.
+//
+
+import Foundation
+import CoreLocation
+
+extension CLLocation {
+
+    /// Calculate distance treating lat and long as unitless Cartesian coordinates
+    func degreeDistance(from: CLLocation) -> Double {
+        let dLat = coordinate.latitude - from.coordinate.latitude
+        let dLon = coordinate.longitude - from.coordinate.longitude
+        return sqrt(dLat * dLat + dLon * dLon)
+    }
+
+    static func geometricMean(_ locations: [CLLocation]) -> CLLocation {
+        var x = 0.0
+        var y = 0.0
+        for location in locations {
+            x += location.coordinate.latitude
+            y += location.coordinate.longitude
+        }
+        x /= Double(locations.count)
+        y /= Double(locations.count)
+        return CLLocation(latitude: x, longitude: y)
+    }
+
+    static func geometricMedian(_ locations: [CLLocation], maxIterations: Int = 200) -> CLLocation {
+        guard locations.count > 0 else {
+            log.error("Empty list. Could not compute median!")
+            return CLLocation(latitude: Double.nan, longitude: Double.nan)
+        }
+
+        guard locations.count > 1 else {
+            return locations.first!
+        }
+
+        // initialize guess to the centroid
+        var median = geometricMean(locations).coordinate
+
+        // if the initial guess is one of the points, the math breaks, so move it
+        while !locations.filter({
+            $0.coordinate.longitude == median.longitude &&
+            $0.coordinate.latitude == median.latitude
+        }).isEmpty {
+            median.latitude += 0.1
```

Comment:
Not sure how often this happens, but 0.1 in lat/long space is up to 11km, so we should add a much smaller number, such as 0.000001, which amounts to a couple of centimetres.